onnx/defs/tensor/old.cc

```diff
@@ -426,40 +426,63 @@ ONNX_OPERATOR_SET_SCHEMA(
             "Constrain input 'X' and output 'Y' to all tensor types.")
         .SetDoc(Upsample_ver9_doc)
         .TypeAndShapeInferenceFunction([](InferenceContext& ctx) {
-          if (!hasNInputShapes(ctx, 1)) {
+          if (!hasNInputShapes(ctx, 2)) {
             return;
           }
           propagateElemTypeFromInputToOutput(ctx, 0, 0);
           auto& input_shape = getInputShape(ctx, 0);
           auto* output_shape = getOutputShape(ctx, 0);
-          output_shape->clear_dim();
           auto scales = ctx.getInputData(1);
+
```

Comment:
If this is more or less just a copy of the above method - maybe we can just shift it to a util.h under`..defs\tensor\` and reference that method in all places. Any concerns with that approach ?

Reply:
Good point. Switch to do so